core/src/test/scala/integration/kafka/api/RackAwareAutoTopicCreationTest.scala

```diff
@@ -37,29 +43,52 @@ class RackAwareAutoTopicCreationTest extends KafkaServerTestHarness with RackAwa
 
   def generateConfigs =
     (0 until numServers) map { node =>
-      TestUtils.createBrokerConfig(node, zkConnect, enableControlledShutdown = false, rack = Some((node / 2).toString))
+      TestUtils.createBrokerConfig(node, zkConnectOrNull, enableControlledShutdown = false, rack = Some((node % 2).toString))
     } map (KafkaConfig.fromProps(_, overridingProps))
 
   private val topic = "topic"
 
-  @Test
-  def testAutoCreateTopic(): Unit = {
-    val producer = TestUtils.createProducer(bootstrapServers())
-    try {
-      // Send a message to auto-create the topic
-      val record = new ProducerRecord(topic, null, "key".getBytes, "value".getBytes)
-      assertEquals(0L, producer.send(record).get.offset, "Should have offset 0")
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testAutoCreateTopic(quorum: String): Unit = {
+    val props = new Properties()
+    props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers())
+
+    TestUtils.waitUntilTrue(
```

Comment:
Do we do this to make sure that we have enough alive brokers for the default replication factor to be satisfied? If there are not enough, does it cause a user-visible error? Just trying to understand why we didn't need it previously.